thriftbp/server.go

```diff
@@ -91,36 +93,145 @@ func NewServer(cfg ServerConfig) (*thrift.TSimpleServer, error) {
 	return server, nil
 }
 
+// NewServerFromOpts creates a new ServerConfig instance and
+// delegates to NewServer so that it can instantiate a
+// new thrift.TSimpleServer.
+func NewServerFromOpts(opts ...ServerOpt) (*thrift.TSimpleServer, error) {
+
+	cfg, err := BuildConfig(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewServer(*cfg)
+}
+
 // NewBaseplateServer returns a new Thrift implementation of a Baseplate
 // server with the given config.
 func NewBaseplateServer(
 	bp baseplate.Baseplate,
 	cfg ServerConfig,
 ) (baseplate.Server, error) {
-	middlewares := BaseplateDefaultProcessorMiddlewares(
-		DefaultProcessorMiddlewaresArgs{
-			EdgeContextImpl:                    bp.EdgeContextImpl(),
-			ErrorSpanSuppressor:                cfg.ErrorSpanSuppressor,
-			ReportPayloadSizeMetricsSampleRate: cfg.ReportPayloadSizeMetricsSampleRate,
-		},
+
+	opts := append(
+		[]ServerOpt{ServerOptFrom(cfg)},
+		DefaultServerOpts(bp)...,
 	)
-	middlewares = append(middlewares, cfg.Middlewares...)
-	cfg.Middlewares = middlewares
-	if cfg.Logger == nil {
+
+	return NewBaseplateServerFromOpts(bp, opts...)
+}
+
+// NewBaseplateServerFromOpts returns a new Thrift implementation of a Baseplate
+// server using a config built from the supplied opts ServerOpt.
+func NewBaseplateServerFromOpts(
+	bp baseplate.Baseplate,
+	opts ...ServerOpt,
+) (baseplate.Server, error) {
+
+	srv, err := NewServerFromOpts(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return ApplyBaseplate(bp, srv), nil
+}
+
+// ServerOpt is a type used for defining configuration arguments
```

Comment:
ServerOption or NewServerOption -- don't drop letters to save typing, especially in exported APIs, and _especially_ when it is ambiguous (Opt can mean Option, Optional, Opt-in, etc).